File: pages/segment_info.js
(function () {
  const response = {
    "type": "page",
    "body": [
      {
        "type": "button",
        "label": "新增管片信息",
        "icon": "fa fa-plus",
        "level": "primary",
        "className": "m-b-sm",
        "actionType": "dialog",
        "dialog": {
          "title": "新增管片信息",
          "body": {
            "type": "form",
            "api": {
              "method": "post",
              "url": "/segment_info/segment_info",
              "messages": {
                "success": "添加成功",
                "failed": "添加失败"
              }
            },
            "body": [
              {
                "type": "input-text",
                "name": "connect_key",
                "label": "盾构机编号",
                "required": true
              },
              {
                "type": "input-number",
                "name": "ring",
                "label": "对应环号",
                "required": true
              },
              {
                "type": "input-text",
                "name": "sub_code",
                "label": "施工区间编码",
                "required": true
              },
              {
                "type": "input-text",
                "name": "sub_name",
                "label": "施工区间名称",
                "required": true
              },
              {
                "type": "input-text",
                "name": "gp_nj",
                "label": "管片内径"
              },
              {
                "type": "input-text",
                "name": "gp_wj",
                "label": "管片外径"
              },
              {
                "type": "input-text",
                "name": "gp_kd",
                "label": "管片宽度"
              },
              {
                "type": "input-text",
                "name": "gp_fks",
                "label": "管片分块数"
              },
              {
                "type": "input-text",
                "name": "dw_sppc",
                "label": "盾尾水平偏差"
              },
              {
                "type": "input-text",
                "name": "ds_szpc",
                "label": "盾尾竖直偏差"
              },
              {
                "type": "input-text",
                "name": "gdj",
                "label": "滚动角"
              },
              {
                "type": "input-text",
                "name": "dg_dta",
                "label": "盾构设计曲线DTA"
              },
              {
                "type": "input-text",
                "name": "gp_pjdw",
                "label": "管片拼接点位"
              },
              {
                "type": "input-text",
                "name": "gp_cql",
                "label": "管片超前量"
              },
              {
                "type": "input-text",
                "name": "gp_zt",
                "label": "管片姿态"
              }
            ]
          }
        }
      },
      {
        "type": "crud",
        "syncLocation": false,
        "resizable": true,
        "columnsTogglable": true,
        "className": "table-db m-b-none",
        "tableClassName": "table-cell-nowrap",
        "api": {
          "url": "/segment_info/segment_info_list",
          "method": "get",
          "dataType": "json",
          "adaptor": function (payload) {
            return {
              status: payload.status,
              msg: payload.msg,
              data: {
                items: payload.data && payload.data.items ? payload.data.items : [],
                total: payload.data && payload.data.total ? payload.data.total : 0
              }
            };
          }
        },
        "headerToolbar": [
          {
            "type": "tpl",
            "tpl": "<div class='header-title'><i class='fa fa-cog'></i> 管片信息管理</div>"
          },
          "export-excel",
          {
            "type": "reload",
            "icon": "fa fa-refresh",
            "tooltip": "刷新数据",
            "tooltipPlacement": "top",
            "level": "link"
          }
        ],
        "filter": {
          "title": "查询条件",
          "submitText": "查询",
          "body": [
            {
              "type": "input-text",
              "name": "connect_key",
              "label": "盾构机编号",
              "placeholder": "请输入盾构机编号进行查询"
            },
            {
              "type": "input-text",
              "name": "sub_code",
              "label": "施工区间编码",
              "placeholder": "请输入施工区间编码进行查询"
            },
            {
              "type": "input-text",
              "name": "sub_name",
              "label": "施工区间名称",
              "placeholder": "请输入施工区间名称进行查询"
            }
          ]
        },
        "columns": [
          {
            "type": "operation",
            "label": "操作",
            "width": 80,
            "fixed": "right",
            "buttons": [
              {
                "type": "button",
                "icon": "fa fa-pencil",
                "tooltip": "修改",
                "level": "link",
                "actionType": "drawer",
                "drawer": {
                  "position": "right",
                  "size": "md",
                  "title": "修改管片信息",
                  "body": {
                    "type": "form",
                    "api": {
                      "method": "post",
                      "url": "/segment_info/update_segment_info",
                      "messages": {
                        "success": "更新成功",
                        "failed": "更新失败"
                      }
                    },
                    "body": [
                      {"type": "hidden", "name": "id"},
                      {"type": "input-text", "name": "connect_key", "label": "盾构机编号", "required": true},
                      {"type": "input-number", "name": "ring", "label": "对应环号", "required": true},
                      {"type": "input-text", "name": "sub_code", "label": "施工区间编码", "required": true},
                      {"type": "input-text", "name": "sub_name", "label": "施工区间名称", "required": true},
                      {"type": "input-text", "name": "gp_nj", "label": "管片内径"},
                      {"type": "input-text", "name": "gp_wj", "label": "管片外径"},
                      {"type": "input-text", "name": "gp_kd", "label": "管片宽度"},
                      {"type": "input-text", "name": "gp_fks", "label": "管片分块数"},
                      {"type": "input-text", "name": "dw_sppc", "label": "盾尾水平偏差"},
                      {"type": "input-text", "name": "ds_szpc", "label": "盾尾竖直偏差"},
                      {"type": "input-text", "name": "gdj", "label": "滚动角"},
                      {"type": "input-text", "name": "dg_dta", "label": "盾构设计曲线DTA"},
                      {"type": "input-text", "name": "gp_pjdw", "label": "管片拼接点位"},
                      {"type": "input-text", "name": "gp_cql", "label": "管片超前量"},
                      {"type": "input-text", "name": "gp_zt", "label": "管片姿态"}
                    ]
                  }
                }
              },
              {
                "type": "button",
                "icon": "fa fa-times text-danger",
                "actionType": "ajax",
                "tooltip": "删除",
                "confirmText": "确定要删除该记录吗？",
                "api": {
                  "method": "delete",
                  "url": "/segment_info/delete_segment_info/${id}"
                },
                "messages": {
                  "success": "删除成功",
                  "failed": "删除失败"
                }
              }
            ]
          },
          { "name": "id","label": "编号","type": "number", "width": 80 },
          { "name": "connect_key", "label": "盾构机编号", "type": "text", "width": 120 },
          { "name": "ring", "label": "对应环号", "type": "number", "width": 100 },
          { "name": "sub_code", "label": "施工区间编码", "type": "text", "width": 120 },
          { "name": "sub_name", "label": "施工区间名称", "type": "text", "width": 150 },
          { "name": "gp_nj", "label": "管片内径", "type": "text", "width": 100 },
          { "name": "gp_wj", "label": "管片外径", "type": "text", "width": 100 },
          { "name": "gp_kd", "label": "管片宽度", "type": "text", "width": 100 },
          { "name": "gp_fks", "label": "管片分块数", "type": "text", "width": 100 },
          { "name": "dw_sppc", "label": "盾尾水平偏差", "type": "text", "width": 120 },
          { "name": "ds_szpc", "label": "盾尾竖直偏差", "type": "text", "width": 120 },
          { "name": "gdj", "label": "滚动角", "type": "text", "width": 100 },
          { "name": "dg_dta", "label": "盾构设计曲线DTA", "type": "text", "width": 150 },
          { "name": "gp_pjdw", "label": "管片拼接点位", "type": "text", "width": 120 },
          { "name": "gp_cql", "label": "管片超前量", "type": "text", "width": 100 },
          { "name": "gp_zt", "label": "管片姿态", "type": "text", "width": 100 }
        ]
      }
    ]
  };
   window.jsonpCallback && window.jsonpCallback(response);
})();

File: api/ORM/api/segment_info.py
from fastapi import APIRouter, Query, HTTPException
from typing import Optional
from segment_models import segment_info
from tortoise.expressions import Q
from pydantic import BaseModel

segment_info_api = APIRouter()

class segmentinfoIn(BaseModel):
    connect_key: str
    ring: int
    sub_code: str
    sub_name: str
    gp_nj: Optional[str] = None
    gp_wj: Optional[str] = None
    gp_kd: Optional[str] = None
    gp_fks: Optional[str] = None
    dw_sppc: Optional[str] = None
    ds_szpc: Optional[str] = None
    gdj: Optional[str] = None
    dg_dta: Optional[str] = None
    gp_pjdw: Optional[str] = None
    gp_cql: Optional[str] = None
    gp_zt: Optional[str] = None

@segment_info_api.post("/segment_info")
async def add_segment_info(info_in: segmentinfoIn):
    try:
        new_info = await segment_info.create(**info_in.dict(exclude_unset=True))
        return {"status": 0, "msg": "添加成功", "data": {"id": new_info.id}}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"添加失败: {str(e)}")

@segment_info_api.get("/segment_info_list")
async def get_segment_info_list(
    id: Optional[int] = Query(None, description="编号"),
    connect_key: Optional[str] = Query(None, description="盾构机编码"),
    sub_code: Optional[str] = Query(None, description="施工区间编码"),
    sub_name: Optional[str] = Query(None, description="施工区间名称"),
    page: int = Query(1, ge=1, description="页码"),
    per_page: int = Query(10, ge=1, le=100, description="每页数量")
):
    filters = Q()
    if id is not None:
        filters &= Q(id=id)
    if connect_key:
        filters &= Q(connect_key=connect_key)
    if sub_code:
        filters &= Q(sub_code=sub_code)
    if sub_name:
        filters &= Q(sub_name=sub_name)

    total = await segment_info.filter(filters).count()
    items = await segment_info.filter(filters).offset((page - 1) * per_page).limit(per_page).all()
    
    # 将数据库对象转换为字典列表
    items_dict = [{
        'id': item.id,
        'connect_key': item.connect_key,
        'ring': item.ring,
        'sub_code': item.sub_code,
        'sub_name': item.sub_name,
        'gp_nj': item.gp_nj,
        'gp_wj': item.gp_wj,
        'gp_kd': item.gp_kd,
        'gp_fks': item.gp_fks,
        'dw_sppc': item.dw_sppc,
        'ds_szpc': item.ds_szpc,
        'gdj': item.gdj,
        'dg_dta': item.dg_dta,
        'gp_pjdw': item.gp_pjdw,
        'gp_cql': item.gp_cql,
        'gp_zt': item.gp_zt
    } for item in items]

    return {
        "status": 0,
        "msg": "获取成功",
        "data": {
            "items": items_dict,
            "total": total,
            "page": page,
            "perPage": per_page,
            "count": total
        }
    }

class SegmentInfoUpdateModel(BaseModel):
    id: int
    connect_key: str
    ring: int
    sub_code: str
    sub_name: str
    gp_nj: Optional[str] = None
    gp_wj: Optional[str] = None
    gp_kd: Optional[str] = None
    gp_fks: Optional[str] = None
    dw_sppc: Optional[str] = None
    ds_szpc: Optional[str] = None
    gdj: Optional[str] = None
    dg_dta: Optional[str] = None
    gp_pjdw: Optional[str] = None
    gp_cql: Optional[str] = None
    gp_zt: Optional[str] = None

@segment_info_api.post("/update_segment_info")
async def update_segment_info(info_data: SegmentInfoUpdateModel):
    try:
        info = await segment_info.get_or_none(id=info_data.id)
        if not info:
            return {"status": 1, "msg": f"未找到ID为{info_data.id}的记录"}
        
        await info.update_from_dict(info_data.dict(exclude_unset=True))
        await info.save()
        return {"status": 0, "msg": "更新成功"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"更新失败: {str(e)}")

@segment_info_api.delete("/delete_segment_info/{id}")
async def delete_segment_info(id: int):
    try:
        info = await segment_info.get_or_none(id=id)
        if not info:
            return {"status": 1, "msg": f"未找到ID为{id}的记录"}
        
        await info.delete()
        return {"status": 0, "msg": "删除成功"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"删除失败: {str(e)}")